17 Custom Middleware/src/middleware/NavigationMware.ts

```diff
@@ -0,0 +1,22 @@
+import * as toastr from 'toastr';
+import {NavigationInfo} from './navigationInfo'
+import { hashHistory } from 'react-router';
+
+function createUINotificationMiddleware() {
```

Comment:
This is just a simple sample, it would be world in a real app to balance the notification needs and cover the most common cases (maybe 80% of the cases could be covered with a middleware like this, then the rest could just be covered using the previous approach we used: flags, component handling that).